container_files/assets/www/doc/builtin/sql/FromExpression.md

```diff
@@ -78,3 +78,54 @@ SELECT x.* FROM merge(dataset1, dataset2, dataset3) AS x
 ```
 
 See ![](%%doclink merged dataset) for more details.
+
+## Using rows as a table
+
+In some circumstances, it may be useful to use a row as a table,
+particularly when using the ![](%%doclink sql.query function) or a
+subtable.  This can be done using the syntax
+
+```
+SELECT ... FROM ROW TABLE (expression) ...
+```
+
+When this construct is used, a table is constructed with one row
+for each column in the expression, with one column called `value`
+containing the value of the column, and the row name containing
+the column name.  For example, the following expression
+
+```
+ROW TABLE ({x: 1, y:2, z: 'three'})
+```
+
+would yield the following table:
+
+```
+rowName() value
+x         1
+y         2
+z         "three"
+```
+
+As an example, the following `sql.query` object would strip
+out any numeric-valued columns and uppercase all names from a
+passed in row:
+
+```
+POST /v1/functions/row_transform {
+    type: 'sql.query',
+    params: {
+        query: 'SELECT upper(rowName()) AS column, value FROM ROW TABLE ($input) WHERE CAST (value AS NUMBER) IS NULL',
```

Comment:
I would prefer syntax that looks more like a function-call like `sample` or `transpose` which we already support. Perhaps `row_table(...)` ?